haystack/core/pipeline/base.py

```diff
@@ -197,12 +197,15 @@ def from_dict(
                         # ...then try again
                         if component_data["type"] not in component.registry:
                             raise PipelineError(
-                                f"Successfully imported module {module} but can't find it in the component registry."
-                                "This is unexpected and most likely a bug."
+                                f"Successfully imported module '{module}' but couldn't find"
+                                f"'{component_data['type']}' in the component registry.\n"
+                                f"The component might be registered under a different path."
+                                f"Here are the components registered:\n {list(component.registry.keys())}\n"
                             )
                     except (ImportError, PipelineError, ValueError) as e:
                         raise PipelineError(
-                            f"Component '{component_data['type']}' (name: '{name}') not imported."
+                            f"Component '{component_data['type']}' (name: '{name}') not imported. Please"
+                            f"check that the package is installed and the component path is correct."
```

Comment:
```suggestion
                            f" check that the package is installed and the component path is correct."
```
we need to add a space otherwise you end up with `Pleasecheck`